src/validatedLogger.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Logger,
+  LogLevel
+} from './types';
+
+type validateLogger = (loglevel: LogLevel) => (message: string, logger: Logger | undefined) => void;
+const validateLogger: validateLogger = (loglevel) => (message, logger) => {
+  if (logger) {
```

Comment:
Maybe we can just return here and don't move whole function into a condition block